### DISCUSSION TITLE
Is System.Text.Json supported?

### DISCUSSION THREAD

**Discussion by xperiandri** [Gremlinq/ExRam.Gremlinq#1079] [Q&A] [answered]
Can I use System.Text.Json instead of Newtonsoft?

**Comment by danielcweber** [comment]
Not out of the box put it is planned as a commercial feature.

**Reply by xperiandri** [reply]
What is the layer required to be implemented?
Could you point to a folder in sources?

**Reply by danielcweber** [reply]
src\Core\Deserialization

**Reply by xperiandri** [reply]
So only 2 converters and we are good?

**Reply by xperiandri** [reply]
1. `SingleItemArrayFallbackConverterFactory`
2. `ToStringFallbackConverterFactory`

**Reply by danielcweber** [reply]
I can't say what it'll take, I haven't tackled this yet.

**Comment by xperiandri** [comment]
@danielcweber as I understand from this code https://github.com/ExRam/ExRam.Gremlinq/blob/05543f936c10cb6145c612dd1ea964dfa4244baa/src/Support.NewtonsoftJson/Extensions/TransformerExtensions.cs#L70
in order to plug in System.Text.Json it is enough to implement all those classes like `DictionaryConverterFactory` used in this file, right?

**Comment by danielcweber** [comment, marked as answer]
This feature is now available as extension library!

https://docs.gremlinq.net/extensions/
